sam_handler/handler.py

```diff
@@ -191,7 +191,11 @@ def handle(self) -> str:
         self._create_generation_destination()
         template_path = self._prepare_template()
         self._build(invoker, template_path)
-        if template_path != self.sam_template_path:
+
+        build_args = self.arguments.get('build_args', {})
```

Comment:
It should not be added to `build_args`, since that's specifically about the arguments passed to `sam build`. Instead, it should be its own parameter.

Reply:
Ack